### DISCUSSION TITLE
Solver and plot configuration data structure design discussion

### DISCUSSION THREAD

**Discussion by j8xixo12** [solvcon/modmesh#405] [Ideas]
Thanks @yungyuc to point out in https://github.com/solvcon/modmesh/pull/391#discussion_r1685662771 that the way to access data of [SolverConfig](https://github.com/solvcon/modmesh/blob/6e634cdb499d4e14a2e0b9dc81c67ed862d4319c/modmesh/app/euler1d.py#L107) and [PlotConfig](https://github.com/solvcon/modmesh/blob/6e634cdb499d4e14a2e0b9dc81c67ed862d4319c/modmesh/app/euler1d.py#L230) is too wordy. 

For exmaple:
https://github.com/solvcon/modmesh/blob/6e634cdb499d4e14a2e0b9dc81c67ed862d4319c/modmesh/app/euler1d.py#L507
https://github.com/solvcon/modmesh/blob/6e634cdb499d4e14a2e0b9dc81c67ed862d4319c/modmesh/app/euler1d.py#L511

I would like to share my thought of this.
There are some items about this data structure of configuration data:
1. Because of our solver and plot configuration is using [QtTableView](https://doc.qt.io/qtforpython-6/PySide6/QtWidgets/QTableView.html) and it's data model is [QtAbstractTableModel](https://doc.qt.io/qtforpython-6/PySide6/QtCore/QAbstractTableModel.html). the `QtTableView` use index-base way to access it's data, for example: `foo[0][1]`
2. For user's perspective, the configuration table has row and column name, using string way to access data is more intuitive, for example : `foo['var1']['value']`

Based on these requirements I think the configuration class should has the following api :
- _ _ getitem _ _ and _ _ setitem _ _ : should support access data by index and string

**Comment by yungyuc** [comment]
Thanks, @j8xixo12 .  If it is doable to enhance to use `__getitem__` and `__setitem__` for index and string data access, how much effort it will take?

**Reply by j8xixo12** [reply]
I think it would take me about 2 hours to implement and test it.
I plan to file a PR in this week.

**Reply by yungyuc** [reply]
If it's just a couple of hours, please go ahead.  It will be efficient to directly discuss in the PR.
